### PR TITLE
Fix gradle build

### PR DESCRIPTION
**Pull request by louiszawadzki**
### What does this PR do?

This PR fixes the `./gradlew build` command and runs tests on Java 17.
There were some `forEach` APIs used that were not available before sdk version 24, so I switched them to for in.

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)
- [ ] If this PR is auto-generated, please make sure also to manually update the code related to the change

### CODE REVIEW

**Review by plousada — APPROVED** [review]

File: packages/core/android/src/main/kotlin/com/datadog/reactnative/DatadogSDKWrapper.kt [hunk 1]
```diff
@@ -42,8 +42,8 @@ object DatadogSDKWrapperStorage {
      * Exposed for testing purposes only.
      */
     fun notifyOnInitializedListeners(ddCore: InternalSdkCore) {
-        onInitializedListeners.forEach {
```
Comment (0xnm) [inline packages/core/android/src/main/kotlin/com/datadog/reactnative/DatadogSDKWrapper.kt:45]: what is the reason for this change? this `forEach` is coming from the Kotlin stdlib (if I'm not wrong) and shouldn't be impacted by the Android stdlib API availability (same for other similar changes).

File: packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdkBridgeExt.kt [hunk 2]
```diff
@@ -11,7 +11,8 @@ import com.facebook.react.bridge.WritableNativeMap
 
 internal fun List<*>.toWritableArray(): WritableNativeArray {
     val list = WritableNativeArray()
-    forEach {
+    for (it in iterator()) {
```
Comment (0xnm) [inline packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdkBridgeExt.kt:14]: with such change it is better to use other name than `it`, because `it` is reserved keyword in Kotlin, and in this scope it is just a variable name, which may be confusing.

File: packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdkBridgeExt.kt [hunk 3]
```diff
@@ -11,7 +11,8 @@ import com.facebook.react.bridge.WritableNativeMap
...
+        @Suppress("NotImplementedDeclaration")
```
Comment (0xnm) [inline packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdkBridgeExt.kt:15]: what is the reason for this declaration?

File: packages/core/android/src/main/kotlin/com/datadog/reactnative/DatadogSDKWrapper.kt (hunk 1 above)
Comment (louiszawadzki) [reply packages/core/android/src/main/kotlin/com/datadog/reactnative/DatadogSDKWrapper.kt:45]: When running `cd packages/core/android &&./gradlew :lintDebug` we would get the following errors for `forEach` calls:

```
/Users/louis.zawadzki/go/src/github.com/DataDog/mobile/dd-sdk-reactnative/packages/core/android/src/main/kotlin/com/datadog/reactnative/DatadogSDKWrapper.kt:45: Error: Call requires API level 24 (current min is 21): java.lang.Iterable#forEach [NewApi]
        onInitializedListeners.forEach {
                               ~~~~~~~
```

File: packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdkBridgeExt.kt (hunk 3 above)
Comment (louiszawadzki) [reply packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdkBridgeExt.kt:15]: This is because we have a `else -> TODO()` at the end of the `when`.

File: packages/core/android/src/main/kotlin/com/datadog/reactnative/DatadogSDKWrapper.kt (hunk 1 above)
Comment (0xnm) [reply packages/core/android/src/main/kotlin/com/datadog/reactnative/DatadogSDKWrapper.kt:45]: this is pretty weird why here the method from JDK is used, because we use `forEach` pretty much a lot in the native SDK where min API is 21 and there are no problems.

The method from Kotlin stdlib is like that:

```
@kotlin.internal.HidesMembers
public inline fun <T> Iterable<T>.forEach(action: (T) -> Unit): Unit {
    for (element in this) action(element)
}
```

I'm pretty sure root cause is somewhere else.
